packages/world/src/WorldProxyFactory.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { Create2 } from "./Create2.sol";
+import { World } from "./World.sol";
+import { IWorldFactory } from "./IWorldFactory.sol";
+import { IBaseWorld } from "./codegen/interfaces/IBaseWorld.sol";
+import { IModule } from "./IModule.sol";
+import { ROOT_NAMESPACE_ID } from "./constants.sol";
+import { WorldProxy } from "./WorldProxy.sol";
+
+/**
+ * @title WorldProxyFactory
+ * @author MUD (https://mud.dev) by Lattice (https://lattice.xyz)
+ * @notice A factory contract to deploy new WorldProxy instances.
+ * @dev This contract allows users to deploy a new World and WorldProxy, install the InitModule, and transfer the ownership.
+ */
+contract WorldProxyFactory is IWorldFactory {
+  /// @notice Address of the init module to be set in the World instances.
+  IModule public immutable initModule;
+  address public immutable worldImplementation;
+
+  /// @param _initModule The address of the init module.
+  constructor(IModule _initModule) {
+    initModule = _initModule;
+
+    // Deploy a world implementation
+    bytes memory worldBytecode = type(World).creationCode;
+    worldImplementation = Create2.deploy(worldBytecode, 0);
```

Comment:
would this revert if there is already a world deployed at this address? ~~i guess it should revert because otherwise the `creator` doesn't match, but maybe we should use the factory address as a salt here to get around the case where this world address already exists? Although it's still possible that someone else deployed with this salt.~~ Could we instead just deploy a world via `new World()` so we don't run into any Create2 conflicts?

Reply:
Yes I agree, that would be simpler also